tests/bdd/test_volume_replicas.py

```diff
@@ -0,0 +1,146 @@
+    """a suitable available pool."""
+    pools = common.get_pools_api().get_pools()
+    assert len(pools) == 2
+
+
+@given("an existing volume")
+def an_existing_volume():
+    """an existing volume."""
+    volume = common.get_volumes_api().get_volume(VOLUME_UUID)
+    assert volume.spec.uuid == VOLUME_UUID
+
+
+@given("the number of volume replicas is greater than one")
+def the_number_of_volume_replicas_is_greater_than_one():
+    """the number of volume replicas is greater than one."""
+    volumes_api = common.get_volumes_api()
+    volume = volumes_api.put_volume_replica_count(VOLUME_UUID, 2)
+    assert volume.spec.num_replicas > 1
+
+
+@when("a user attempts to decrease the number of volume replicas")
+def a_user_attempts_to_decrease_the_number_of_volume_replicas(replica_ctx):
+    """a user attempts to decrease the number of volume replicas."""
+    volumes_api = common.get_volumes_api()
+    volume = volumes_api.get_volume(VOLUME_UUID)
+    num_replicas = volume.spec.num_replicas
+    volume = volumes_api.put_volume_replica_count(VOLUME_UUID, num_replicas - 1)
+    replica_ctx[REPLICA_CONTEXT_KEY] = volume.spec.num_replicas
+
+
+@when("a user attempts to increase the number of volume replicas")
+def a_user_attempts_to_increase_the_number_of_volume_replicas(replica_ctx):
+    """a user attempts to increase the number of volume replicas."""
+    volumes_api = common.get_volumes_api()
+    volume = volumes_api.get_volume(VOLUME_UUID)
+    num_replicas = volume.spec.num_replicas
+    volume = volumes_api.put_volume_replica_count(VOLUME_UUID, num_replicas + 1)
+    replica_ctx[REPLICA_CONTEXT_KEY] = volume.spec.num_replicas
+
+
+@then("a replica should be removed from the volume")
+def a_replica_should_be_removed_from_the_volume(replica_ctx):
+    """a replica should be removed from the volume."""
+    volume = common.get_volumes_api().get_volume(VOLUME_UUID)
+    assert len(volume.state.children) > 0
+    nexus = volume.state.children[0]
+    assert replica_ctx[REPLICA_CONTEXT_KEY] == len(nexus.children)
+
+
+@then("an additional replica should be added to the volume")
+def an_additional_replica_should_be_added_to_the_volume(replica_ctx):
+    """an additional replica should be added to the volume."""
+    volume = common.get_volumes_api().get_volume(VOLUME_UUID)
+    assert len(volume.state.children) > 0
+    nexus = volume.state.children[0]
+    assert replica_ctx[REPLICA_CONTEXT_KEY] == len(nexus.children)
+
+
+@then("setting the number of replicas to zero should fail with a suitable error")
+def setting_the_number_of_replicas_to_zero_should_fail_with_a_suitable_error():
+    """the replica removal should fail with a suitable error."""
+    volumes_api = common.get_volumes_api()
+    volume = volumes_api.get_volume(VOLUME_UUID)
+    assert len(volume.state.children) > 0
+    try:
+        volumes_api.put_volume_replica_count(VOLUME_UUID, 0)
+    except Exception as e:
+        assert "ApiValueError" in str(type(e))
```

Comment:
should we add a comment here to revisit and try to trigger the proper error?

Reply:
Done.